CT-1830 ensure we only retrieve accepted responder assignments

misaka opened the pull request:
We had a bug where in some situations, when retrieving "responder_assignment" on
a case we would return an assignment that hadn't been accepted, which led to
errors when trying to, for example, trying to send an email to the responding
user assigned to a case.

misaka commented on app/policies/case_policy.rb:
```diff
@@ -64,9 +64,8 @@ def can_accept_or_reject_responder_assignment?
-        check_user_is_an_approver_for_case
-
+    check_user_is_not_a_manager &&
+      ( check_user_is_in_current_team || check_user_is_an_approver_for_case)
```
Doesn't this mean it'll show up when a responder hasn't accepted the case yet? Is that an issue?

stephenrichards replied:
is there a user of the same team as the current user assigned to the case

misaka commented on spec/controllers/assignments_controller/reassign_user_spec.rb:
```diff
@@ -1,19 +1,20 @@
-  let(:assigned_case)     { create :assigned_case }
-  let(:assigned_case_trigger) { create :assigned_case, :flagged_accepted,
+  let(:accepted_case)     { create :accepted_case }
+  # let(:assigned_case)     { create :assigned_case }
```
Looks like we don't need this anymore.

misaka committed:
CT-1830 only send notification to the assigned responding team

Fixes bug where for cases with multiple responder assignments, some still in
pending, we tried to send an email to the pending assignment, raising an error
in the sidekiq process.
